supervisor/api/supervisor.py

```diff
@@ -155,27 +153,15 @@ async def options(self, request: web.Request) -> None:
         if ATTR_FORCE_SECURITY in body:
             self.sys_security.force = body[ATTR_FORCE_SECURITY]
 
-        if ATTR_ADDONS_REPOSITORIES in body:
-            new = set(body[ATTR_ADDONS_REPOSITORIES])
-            await asyncio.shield(self.sys_store.update_repositories(new))
-
-            # Fix invalid repository
-            found_invalid = False
-            for suggestion in self.sys_resolution.suggestions:
-                if (
-                    suggestion.type != SuggestionType.EXECUTE_REMOVE
-                    and suggestion.context != ContextType
-                ):
-                    continue
-                found_invalid = True
-                await self.sys_resolution.apply_suggestion(suggestion)
-
-            if found_invalid:
-                raise APIError("Invalid Add-on repository!")
-
+        # Save changes before processing addons in case of errors
         self.sys_updater.save_data()
         self.sys_config.save_data()
 
+        if ATTR_ADDONS_REPOSITORIES in body:
+            await asyncio.shield(
+                self.sys_store.update_repositories(set(body[ATTR_ADDONS_REPOSITORIES]))
+            )
+
```

Comment:
Ditch that, it is not needed anymore. We can update the UI + CLI with this update too. Still have the same issue then

Reply:
You want to toss the existing API as part of this? That makes this a much bigger breaking change. I guess I was thinking to separate the frontend changes into another PR that included removing/deprecating this API. I can do it all at once though I suppose.